module/VuFind/src/VuFind/Search/Base/Results.php

```diff
@@ -881,6 +881,18 @@ protected function buildFacetList(array $facetList, array $filter = null): array
                     'isApplied'
                 );
             }
+            if ($hierarchical && method_exists($this->hierarchicalFacetHelper, 'filterFacets')) {
+                $excludeFilters = $this->getOptions()->getHierarchicalExcludeFilters();
```

Comment:
Might it make sense to make $field a parameter to the getters? It could be optional (if null, return all values), but if provided it could return just the requested field, and that could make some of the subsequent code a little more concise.